demo/workflows/human-in-the-loop.ts

```diff
@@ -58,16 +65,26 @@ For example, alex is from "Alexander the Great", who was a king of the ancient G
   if (tools && tools.length > 0) {
     const askName = tools.find((tool) => tool.function.name === "ask_name");
     if (askName) {
-      return request(humanInteractionEvent, askName.function.arguments);
+      const snapshot = await getContext().snapshot();
+      return humanInteractionRequestEvent.with({
+        reason: askName.function.arguments,
+        snapshot,
+      });
```

Comment:
good idea, but I'm worry if this will leak the implementation into the workflow

Reply:
you could create the snapshot automatically for this event type, e.g.:
```
const humanInteractionRequestEvent = withSnapshot(workflowEvent<{
  reason: string;
}>());
```
but i think it adds more complexity.

Reply:
I feel like calling `snapshot` in the handler inside is a bad idea

Reply:
Don't see a big problem but you can call it implicitly for an event wrapped with withSnapshot